src/libxrpl/protocol/STTx.cpp

```diff
@@ -187,6 +189,23 @@ STTx::getSeqProxy() const
     if (seq != 0)
         return SeqProxy::sequence(seq);
 
+    if (isFieldPresent(sfBatchTxn))
+    {
+        STObject const batchTxn = const_cast<ripple::STTx&>(*this)
+                                      .getField(sfBatchTxn)
+                                      .downcast<STObject>();
+
+        std::uint32_t const batchIndex{batchTxn.getFieldU8(sfBatchIndex)};
+        if (batchTxn.isFieldPresent(sfTicketSequence))
+        {
+            std::uint32_t const ticketSeq{
+                batchTxn.getFieldU32(sfTicketSequence)};
+            return SeqProxy{SeqProxy::ticket, ticketSeq + batchIndex};
+        }
+        std::uint32_t const startSequence{batchTxn.getFieldU32(sfSequence)};
+        return SeqProxy::sequence(startSequence + batchIndex);
```

Comment:
Wouldn't this cause some sequences to be skipped? 
Ex. 
Inner transactions:
1. alice's tx with sequence 5, with batch index 0
2. bob's tx with sequence 10, with batch index 1
3. alice's tx with sequence 5, with batch index 2

From the spec:

>This is the next valid sequence number of the inner transaction sender (i.e. all inner and outer transactions from the same account will have the same BatchTxn.Sequence value)

It says `Sequence` is the same for the same account, therefore Alice's sequence in all her inner transactions is 5.

But, for the first transaction, the function returns 5 as the sequence. And for the third transaction, the function returns 7 as the sequence. And it seems like the sequence of 6 is skipped.

cc @mvadari

Reply:
Yes.